Mathlib/AlgebraicTopology/SimplexCategory.lean

```diff
@@ -239,6 +265,54 @@ def mkOfLeComp {n} (i j k : Fin (n + 1)) (h₁ : i ≤ j) (h₂ : j ≤ k) :
       | 0, 2, _ => Fin.le_trans h₁ h₂
   }
 
+/-- The "inert" morphism associated to a subinterval `j ≤ i ≤ j + l` of `Fin (n + 1)`.-/
+def subinterval {n} (j l : ℕ) (hjl : j + l ≤ n) :
+    ([l] : SimplexCategory) ⟶ [n] :=
+  SimplexCategory.mkHom {
+    toFun := fun i => ⟨i.1 + j, (by omega)⟩
+    monotone' := fun i i' hii' => by simpa only [Fin.mk_le_mk, add_le_add_iff_right] using hii'
+  }
+
+@[simp]
+lemma const_subinterval_eq {n} (j l : ℕ) (hjl : j + l ≤ n) (i : Fin (l + 1)) :
+    [0].const [l] i ≫ subinterval j l hjl =
+    [0].const [n] ⟨j + i.1, lt_add_of_lt_add_right (Nat.add_lt_add_left i.2 j) hjl⟩  := by
+  unfold subinterval
+  have := const_comp [0] (subinterval j l hjl) i
+  convert this
+  unfold subinterval
+  simp only [len_mk, mkHom, Hom.toOrderHom_mk, OrderHom.coe_mk]
+  exact Nat.add_comm j i.1
```

Comment:
```suggestion
  rw [const_comp] 
  congr
  ext
  dsimp [subinterval]
  rw [add_comm]
```

Reply:
Thank you. I thought this shouldn't be this hard.